addons/compat_spe/compat_spe_refuel/CfgVehicles.hpp

```diff
@@ -1,12 +1,57 @@
 class CfgVehicles {
-    class SPE_Halftrack_base;
-    class SPE_US_M3_Halftrack_Fuel: SPE_Halftrack_base {
-        EGVAR(refuel,hooks)[] = {{-0.23,-2.58,-0.59}};
-        EGVAR(refuel,fuelCargo) = 2000;
-    };
-    class SPE_OpelBlitz_base;
-    class SPE_OpelBlitz_Fuel: SPE_OpelBlitz_base {
-        EGVAR(refuel,hooks)[] = {{-0.23,-2.58,-0.59}};
-        EGVAR(refuel,fuelCargo) = 2000;
+    // Vehicle animation interactions
+    class SPE_Car_base;
+    class SPE_G503_MB_base: SPE_Car_base {
+        class EGVAR(interaction,anims) {
+            class hide_jerry_can_source {
+                positions[] = {"_target selectionPosition ['hide_jerry_can', 'ViewGeometry', 'AveragePoint']"};
+                items[] = {"Land_CanisterFuel_F"};
+                name = ECSTRING(refuel,TakeFuelCanister);
+                text = ECSTRING(refuel,TakeFuelCanisterAction);
+            };
+        };
+    };
+
+    class SPE_Truck_base;
+    class SPE_CCKW_353_Base: SPE_Truck_base {
+        class EGVAR(interaction,anims) {
+            class spare_fuel_hide_source {
+                positions[] = {{0.8, 1.8, -1}, {-0.8, 1.8, -1}, {0.8, 3.5, -1}, {-0.8, 3.5, -1}};
+                items[] = {"Land_CanisterFuel_F", "Land_CanisterFuel_F", "Land_CanisterFuel_F", "Land_CanisterFuel_F", "Land_CanisterFuel_F", "Land_CanisterFuel_F"};
+                name = ECSTRING(refuel,TakeFuelCanister);
+                text = ECSTRING(refuel,TakeFuelCanisterAction);
+            };
+        };
+    };
+    class SPE_CCKW_353_Ammo: SPE_CCKW_353_Base {
+        class EGVAR(interaction,anims): EGVAR(interaction,anims) {
+            class spare_fuel_hide_source: spare_fuel_hide_source {
+                positions[] = {{0.8, 1.8, -0.85}, {-0.8, 1.8, -0.85}, {0.8, 3.5, -0.85}, {-0.8, 3.5, -0.85}};
+            };
+        };
+    };
+    class SPE_CCKW_353_Repair: SPE_CCKW_353_Base {
+        class EGVAR(interaction,anims): EGVAR(interaction,anims) {
+            class spare_fuel_hide_source: spare_fuel_hide_source {
+                positions[] = {{0.8, 1.9, -0.85}, {-0.8, 1.9, -0.85}, {0.8, 3.6, -0.85}, {-0.8, 3.6, -0.85}};
+            };
+        };
+    };
+    class SPE_CCKW_353_Fuel: SPE_CCKW_353_Base {
+        class EGVAR(interaction,anims): EGVAR(interaction,anims) {
+            class spare_fuel_hide_source: spare_fuel_hide_source {
+                positions[] = {{0.8, 1.25, -0.8}, {-0.8, 1.25, -0.8}, {0.8, 2.95, -0.8}, {-0.8, 2.95, -0.8}, {0.8, -1, -0.3}, {-0.8, -1, -0.3}};
+                items[] = { // 32x
+                    "Land_CanisterFuel_F", "Land_CanisterFuel_F", "Land_CanisterFuel_F", "Land_CanisterFuel_F",
```

Comment:
I'm worried about physx collisions,
we don't have to be exact on the number imho

Reply:
`EFUNC(common,findUnloadPosition)` handles finding an empty space. I'll test this more, but I don't think there's much of a reason to be worried.

Reply:
ace_interaction_fnc_initAnimActions:
```
            // If 1 object was spawned in, consider it a success
```
also code will try to spawn as many as it can but won't fail if it can't find room for all of them

Reply:
Yea, I changed it to work like that, because I'm not sure what to consider a success otherwise.